### ISSUE TITLE
Check types of builtins

### ISSUE THREAD

**Issue by fblanqui** [Deducteam/lambdapi#196]
There is currently no check done on builtins. It is important to give some feed back to the user when he sets some builtins.

**Comment by rlepigre** [comment]
What kind of feedback would you expect? Builtins are only used to register a symbol under a specific, globally known name. Of course, tactics like `rewrite` complain if the required builtin symbols have not been defined, and the same goes for natural number literals.

**Comment by fblanqui** [comment]
In rewrite for instance, it's easy to make a mistake by permuting some arguments, or by taking an argument of type P(x=y) instead of P(y=x). It's important to detect this as soon as possible. Otherwise, people are not going to understand why the rewrite tactic fails.

**Comment by fblanqui** [comment]
See https://github.com/Deducteam/lambdapi/blob/master/doc/sections/tactics.md. The types of constants are quite constrained:
```
symbol T : U ⇒ TYPE
symbol P : Prop ⇒ TYPE
symbol eq {a} : T a ⇒ T a ⇒ Prop
symbol refl {a} (x:T a) : P (x = x)
symbol eqind {a} x y : P (x = y) ⇒ ∀(p:T a⇒Prop), P (p y) ⇒ P (p x)
```

**Comment by rlepigre** [comment]
Yeah but it is not possible to do that in a generic way... As builtins will only be instantiated once, and almost always by the library part defining the logic, I don't see that as a big problem. I don't think users will get to manipulate builtins that often directly, and if they mess up doing that then that is their problem.

**Comment by fblanqui** [comment]
Fixed in 93b39e515ab607dcc94a117f979fc8cf40882881 for rewrite.
